aip/0216.md

```diff
@@ -60,13 +60,57 @@ The field referencing the `State` enum in a resource **should** behave and be
 documented as "Output only", in accordance with [AIP-203](./0203.md).
 
 APIs **should not** allow a `State` enum to be directly updated through an
-"update" method, and **should** instead use a custom method.
+"update" method, and **should** instead use a custom state transition method.
 
 This is because update methods are generally not expected to have side effects,
 and also because updating state directly implies that it is possible to set the
 state to any available value, whereas states generally reflect a resource's
 progression through a lifecycle.
 
+### State transition methods
+
+State transition methods should look like this:
+
+```proto
+// Publishes a book.
+// The `state` of the book after publishing will be PUBLISHED.
+rpc PublishBook(PublishBookRequest) returns (Book) {
+  option (google.api.http) = {
+    post: "/v1/{name=publishers/*/books/*}:publish"
+    body: "*"
+  };
+}
+```
+
+- The name of the method **should** be a verb followed by the singular form of
+  the resource's message name.
+- The request message **must** match the RPC name, with a `-Request` suffix.
+- The response message **must** be the resource itself.
+- The HTTP verb **must** be `POST`.
+- The HTTP URI **must** use a `:` character followed by the custom verb
+  (`:publish` in the above example), and the verb in the URI **must** match the
+  verb in the name of the RPC.
+  - If word separation is required, `camelCase` **must** be used.
+- The `body` clause in the `google.api.http` annotation **must** be `"*"`.
+- The request message field receiving the resource name **should** map to the
+  URL path.
+  - This field **should** be called `name`.
+  - The `name` field **should** be the only variable in the URI path. All
+    remaining parameters **should** map to URI query parameters.
+
+The request message should look like this:
+
+```proto
+message PublishBookRequest {
+  // The name of the book to publish.
+  // Format: publishers/{publisher}/books/{book}
+  string name = 1;
+}
+```
+
+- A resource name field **must** be included. It **should** be called `name`.
+- The comment for the field **should** document the resource pattern.
+
```

Comment:
We should have some addition guidance on error cases (when you try to publish a book that's already published, do you return an error? or do you just assume that the work was done?)

Reply:
Added